src/interp/interp-wasi.cc

```diff
@@ -76,18 +120,247 @@ class WasiInstance {
         uvwasi(uvwasi),
         memory(memory) {}
 
+  Result random_get(const Values& params,
+                             Values& results,
+                             Trap::Ptr* trap) {
+    /* __wasi_errno_t __wasi_random_get(uint8_t * buf, __wasi_size_t buf_len) */
+    assert(false);
+    return Result::Ok;
+  }
+
   Result proc_exit(const Values& params, Values& results, Trap::Ptr* trap) {
     const Value arg0 = params[0];
     uvwasi_proc_exit(uvwasi, arg0.i32_);
     return Result::Ok;
   }
 
+  Result poll_oneoff(const Values& params, Values& results, Trap::Ptr* trap) {
+    assert(false);
+    return Result::Ok;
+  }
+
+  Result clock_time_get(const Values& params,
+                        Values& results,
+                        Trap::Ptr* trap) {
+    /* __wasi_errno_t __wasi_clock_time_get(__wasi_clockid_t id,
+     *                                      __wasi_timestamp_t precision,
+     *                                      __wasi_timestamp_t *time)
+     */
+    __wasi_timestamp_t t;
+    results[0].i32_ =
+        uvwasi_clock_time_get(uvwasi, params[0].i32_, params[1].i64_, &t);
+    uint32_t time_ptr = params[2].i32_;
+    CHECK_RESULT(writeValue<__wasi_timestamp_t>(t, time_ptr, trap));
+    return Result::Ok;
+  }
+
+  Result path_rename(const Values& params, Values& results, Trap::Ptr* trap) {
+    assert(false);
+    return Result::Ok;
+  }
+
+  Result path_open(const Values& params, Values& results, Trap::Ptr* trap) {
+    /* __wasi_errno_t __wasi_path_open(__wasi_fd_t fd,
+                                       __wasi_lookupflags_t dirflags,
+                                       const char *path,
+                                       size_t path_len,
+                                       __wasi_oflags_t oflags,
+                                       __wasi_rights_t fs_rights_base,
+                                       __wasi_rights_t fs_rights_inherting,
+                                       __wasi_fdflags_t fdflags,
+                                       __wasi_fd_t *opened_fd) */
+    uvwasi_fd_t dirfd = params[0].i32_;
```

Comment:
OK for now, but we'll want to check the number of params before accessing them, add a TODO?

Reply:
This change adds such a TODO in WasiBindImports (which is where I think we would want to check such things right?